lib/Dialect/TTNN/Transforms/Optimizer.cpp

```diff
@@ -234,6 +216,18 @@ class TTNNOptimizer : public impl::TTNNOptimizerBase<TTNNOptimizer> {
     assert(overrideInputLayout.size() == overrideReshardEdges.size());
   }
 
+  mlir::TypedValue<mlir::tt::DeviceType>
+  getDeviceOpValue(Operation *contextOp) {
+    Block *block = contextOp->getBlock();
+    mlir::TypedValue<mlir::tt::DeviceType> deviceOpResult;
+    for (auto &op : block->getOperations()) {
+      if (GetDeviceOp deviceOp = dyn_cast<GetDeviceOp>(op)) {
+        deviceOpResult = deviceOp.getResult();
```

Comment:
Could you return here or do you need to find the "last" device op?

Reply:
I think we can have only one. Adding break.